crates/sui-graphql-rpc/src/client/simple_client.rs

```diff
@@ -148,3 +152,29 @@ pub fn resolve_variables(
 
     Ok((type_defs, var_vals))
 }
+
+pub const fn is_valid_variable_name_char(c: char) -> bool {
+    matches!(c, '_' | 'a'..='z' | 'A'..='Z' | '0'..='9')
+}
+
+/// Returns `true` if all bytes in `b` after the offset `start_offset` are valid
+/// ASCII variable name characters.
+const fn all_bytes_valid(b: &[u8], start_offset: usize) -> bool {
+    let mut i = start_offset;
+    while i < b.len() {
+        if !is_valid_variable_name_char(b[i] as char) {
+            return false;
+        }
+        i += 1;
+    }
+    true
+}
+
+pub const fn is_valid_variable_name(s: &str) -> bool {
+    let b = s.as_bytes();
+    match b {
+        [b'a'..=b'z', ..] | [b'A'..=b'Z', ..] => all_bytes_valid(b, 1),
+        [b'_', ..] if b.len() > 1 => all_bytes_valid(b, 1),
+        _ => false,
+    }
+}
```

Comment:
nits on naming here -- `all_bytes_valid` is quite a generic name, when this function is checking specifically that the bytes are a valid continuation of a GraphQL variable name, similarly, `is_valid_variable_name_char` is [more properly `is_valid_variable_name_continue`](https://github.com/graphql/graphql-spec/blob/main/spec/Section%202%20--%20Language.md#names).

You can side step the naming issue entirely by inlining everything into one function, which ends up quite neat if you use the `chars` and `all` functions.  This may not be a `const` function, but I don't think that's a thing we should care about for this function:

```suggestion
pub const fn is_valid_variable_name(s: &str) -> bool {
    let mut cs = s.chars();
    let Some(fst) = cs.next() else { return false };
    
    match fst {
      '_' =>  if s.len() > 1 { /* nop */ };
      'a' ..= 'z' | 'A' ..= 'Z'  => { /* nop */ }
      _ => return false
    }
    
    cs.all(|c| matches!(c, '_' | 'a' ..= 'z' | 'A' ..= 'Z' | '0' ..= '9')
}
```